Mathlib/Topology/ContinuousFunction/CFC.lean

```diff
@@ -0,0 +1,1111 @@
+
+-- Question: do we want to make this a `UniformEmbedding` too?
+/-- A `cfc_class R a` is a star algebra homomorphism from the continuous
+`R`-valued functions defined on the spectrum of `a : A` into the algebra `A` which is in addiiton
+continuous and extends the polynomial functional calculus. More precisely, this latter statement
+is encapsulated in `cfc₁_map_polynomial`. -/
+@[ext (flat := false)]
+class CFCClass (R : Type*) {A : Type*} [CommSemiring R] [StarRing R] [TopologicalSpace R]
+    [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A] [TopologicalSpace A]
+    [Algebra R A] (a : A) extends C(spectrum R a, R) →⋆ₐ[R] A where
+  /-- A continuous functional calculus is a closed embedding. -/
+  hom_closedEmbedding : ClosedEmbedding toStarAlgHom
+  /-- A continuous functional calculus extends the polynomial functional calculus. -/
+  hom_map_X : toStarAlgHom (toContinuousMapOnAlgHom (spectrum R a) X) = a
+  /-- A continuous functional calculus satisfies the spectral mapping property. -/
+  hom_map_spectrum : ∀ f, spectrum R (toStarAlgHom f) = Set.range f
+
+-- see note [lower instance priority]
+instance (priority := 100) CFCClass.toCFCCoreClass {R A : Type*} [CommSemiring R] [StarRing R]
+    [TopologicalSpace R] [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A]
+    [TopologicalSpace A] [Algebra R A] (a : A) [CFCClass R a] : CFCCoreClass R a :=
+  { (‹_› : CFCClass R a) with
+    hom_continuous := (‹_› : CFCClass R a).hom_closedEmbedding.continuous }
+
+-- see note [lower instance priority]
+instance (priority := 100) CFCClass.subsingleton {R A : Type*} [CommSemiring R] [StarRing R]
+    [TopologicalSpace R] [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A]
+    [TopologicalSpace A] [Algebra R A] (a : A) [Subsingleton (CFCCoreClass R a)] :
+    Subsingleton (CFCClass R a) :=
+  have : Function.Injective (@CFCClass.toCFCCoreClass R _ _ _ _ _ _ _ _ _ _ a ·) :=
+    fun _ _ h => CFCClass.ext _ _ <| (CFCCoreClass.ext_iff _ _).mp h
+  this.subsingleton
+
+section generic
+
+section Semiring
+
+variable (R : Type*) {A : Type*} [CommSemiring R]
+    [StarRing R] [TopologicalSpace R] [TopologicalSemiring R] [ContinuousStar R] [Ring A]
+    [StarRing A] [TopologicalSpace A] [Algebra R A] (a : A)
+
+section CFCCore
+
+variable [CFCCoreClass R a]
+
+/-- The `StarAlgHom` underlying an instance of the continuous functional calculus. -/
+def cfc₁ : C(spectrum R a, R) →⋆ₐ[R] A := CFCCoreClass.toStarAlgHom
+
+/-- This is `cfc₁` composed with the natural star algebra homomorphism from `C(R, R)` into
+`C(spectrum R a, R)` given by precompostion with the embedding of `spectrum R a` into `R`.
+
+While `cfc₁` is necessary in order to have some of the key properties (e.g., uniqueness of the
+continuous funcitonal calculus, injectivity, mapping into the `elementralStarAlgebra`, etc.), it
+is expected that this version will be more useful in practice. In particular, it will naturally
```

Comment:
```suggestion
continuous funcitonal calculus, injectivity, mapping into the `elementalStarAlgebra`, etc.), it
is expected that this version will be more useful in practice. In particular, it will naturally
```